### PR TITLE
fix(product): add sorting options

### PR DESCRIPTION
**Pull request by IvanMaidurov**

### CODE REVIEW

File: product/infrastructure/fake/searchservice.go [hunk 1]
```diff
@@ -168,17 +182,26 @@ func (s *SearchService) findCurrentPage(filters []searchDomain.Filter) int {
...
-	// first check if the searchDomain.SortFilter is given
-	for _, v := range filters {
-		if filter, ok := v.(*searchDomain.SortFilter); ok {
-			result = append(result, searchDomain.SortOption{
-				Label:        filter.Field(),
-				Field:        filter.Field(),
-				SelectedAsc:  !filter.Descending(),
-				SelectedDesc: filter.Descending(),
-			})
+// mapSortOptions maps searchperience delivered sort options to corresponding domain objects.
```
Comment (carstendietrich) [inline product/infrastructure/fake/searchservice.go:185]:
```suggestion
// mapSortOptions maps configured sort options to the ones potentially provided via filter
```

File: product/infrastructure/fake/searchservice_test.go [hunk 2]
```diff
@@ -258,10 +270,28 @@ func TestSearchService_SearchBy(t *testing.T) {
...
+						},
+						testSortConfig{
+							Key:   "size",
+							Label: "size",
+							Desc:  "size",
+						},
+						testSortConfig{
+							Key:   "no-direction",
+							Label: "no-direction",
+							Asc:   "no-direction",
```
Comment (carstendietrich) [inline product/infrastructure/fake/searchservice_test.go:292]: can we have an example with both asc/desc set?

File: product/infrastructure/fake/searchservice.go [hunk 3]
```diff
@@ -17,27 +19,39 @@ type (
...
 
 // Inject dependencies
 func (s *SearchService) Inject(
 	productService *ProductService,
 	cfg *struct {
-		LiveSearchJSON         string `inject:"config:commerce.product.fakeservice.jsonTestDataLiveSearch,optional"`
-		CategoryFacetItemsJSON string `inject:"config:commerce.product.fakeservice.jsonTestDataCategoryFacetItems,optional"`
+		LiveSearchJSON         string       `inject:"config:commerce.product.fakeservice.jsonTestDataLiveSearch,optional"`
+		CategoryFacetItemsJSON string       `inject:"config:commerce.product.fakeservice.jsonTestDataCategoryFacetItems,optional"`
+		SortConfig             config.Slice `inject:"config:commerce.product.fakeservice.sorting"`
```
Comment (carstendietrich) [inline product/infrastructure/fake/searchservice.go:46]: we should put something to the cueconfig of the module

File: product/infrastructure/fake/searchservice.go [hunk 4]
```diff
@@ -17,27 +19,39 @@ type (
...
 	},
 ) *SearchService {
 	s.productService = productService
 	if cfg != nil {
 		s.liveSearchJSON = cfg.LiveSearchJSON
 		s.categoryFacetItemsJSON = cfg.CategoryFacetItemsJSON
+
+		_ = cfg.SortConfig.MapInto(&s.sortConfig)
```
Comment (carstendietrich) [inline product/infrastructure/fake/searchservice.go:54]: we could panic if an error occurs

File: search/domain/service.go [hunk 5]
```diff
@@ -64,10 +64,10 @@ type (
 		SelectedDesc bool
 		// Asc - represents the field that is used to trigger ascending search.
 		// Deprecated: use "Field" and "SelectedAsc" instead to set which field should be sortable
-		Asc string
+		Asc string // Should it be deprecated ?? marked as deprecated in 2019 and used in 2020
```
Comment (carstendietrich) [inline search/domain/service.go:67]: no hard opinion on that, maybe we have a chat with more contributors on this.

**Review by carstendietrich — CHANGES_REQUESTED** [review]

File: product/infrastructure/fake/searchservice.go (hunk 1 above)
Comment (IvanMaidurov) [reply product/infrastructure/fake/searchservice.go]: fixed

File: product/infrastructure/fake/searchservice_test.go (hunk 2 above)
Comment (IvanMaidurov) [reply product/infrastructure/fake/searchservice_test.go:533]: added

File: product/infrastructure/fake/searchservice.go (hunk 3 above)
Comment (IvanMaidurov) [reply product/infrastructure/fake/searchservice.go:46]: done

File: product/infrastructure/fake/searchservice.go (hunk 4 above)
Comment (IvanMaidurov) [reply product/infrastructure/fake/searchservice.go]: done

File: search/domain/service.go (hunk 5 above)
Comment (carstendietrich) [reply search/domain/service.go:67]: since nobody seems to care feel free to remove your comment and the deprecation notice.

**Review by carstendietrich — CHANGES_REQUESTED** [review]

**Review by carstendietrich — APPROVED** [review]

**Pull request merged**